adapta/storage/blob/s3_storage_client.py

```diff
@@ -31,22 +32,27 @@
 
 
 @final
-class S3StorageClient(StorageClient, ABC):
+class S3StorageClient(StorageClient):
     """
     S3 Storage Client.
     """
 
-    def __init__(self, *, base_client: AwsClient):
+    def __init__(self, *, base_client: AwsClient, endpoint_url: Optional[str] = None):
         super().__init__(base_client=base_client)
         if base_client.session is None:
             raise ValueError("AwsClient.initialize_session should be called before accessing S3StorageClient")
```

Comment:
question to @s-vitaliy here - why don't we call initialize inside this client? To avoid side-effects?

Reply:
Yes, It's a bad practice to call an external services in a class constructor. Not sure if the `base_client.session.resource("s3")`  call below invokes Amazon API, if it is, we should move it out of constructor as well.

Reply:
In that case better create a static constructor: S3StorageClient.create - that does both?

Reply:
I've moved the `base_client.session.resource("s3")` out of the constructor in [26be4c2](https://github.com/SneaksAndData/adapta/pull/437/commits/26be4c28516d5b85ce5f8855d05f2f0913a6bd86) to a new method which also adds and extra set-up step when using the storage client which I'm wondering if it's acceptable. Also updated the ReadMe's accordingly.